### ISSUE TITLE
Measuring the Effects of Data Parallelism on Neural Network Training

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#403]
## Links
https://arxiv.org/abs/1811.03600
https://youtu.be/huTx3rtv8q8?si=4ehe1CS0V5Z986PG

## Authors
[Christopher J. Shallue](https://arxiv.org/search/cs?searchtype=author&query=Shallue,+C+J), [Jaehoon Lee](https://arxiv.org/search/cs?searchtype=author&query=Lee,+J), [Joseph Antognini](https://arxiv.org/search/cs?searchtype=author&query=Antognini,+J), [Jascha Sohl-Dickstein](https://arxiv.org/search/cs?searchtype=author&query=Sohl-Dickstein,+J), [Roy Frostig](https://arxiv.org/search/cs?searchtype=author&query=Frostig,+R), [George E. Dahl](https://arxiv.org/search/cs?searchtype=author&query=Dahl,+G+E)

## Overview
- 本論文は、大規模データ並列学習における バッチサイズと学習ステップ数の関係 を包括的に実証した。
- 実験により、バッチサイズ拡大は当初は学習ステップ数を比例的に削減するが、やがて 逓減効果 (diminishing returns) を経て限界に達することを示した。
- これはモデル・最適化手法・データセットに依存するが、普遍的なスケーリング関係 が存在することを大規模実験で明らかにした。

## 研究の背景と目的
- 深層学習では データ並列化 (data parallelism) が学習の高速化に重要。
- しかし「大きなバッチサイズは汎化性能を劣化させるのか？」や「どの程度までバッチサイズを増やせば有効か？」については過去研究で見解が分かれていた。
- 本論文は、これを大規模かつ体系的に検証する初めての試み。71,638,836 の loss 計測値を含む 168,160 モデル (35 workload) を実験した点が新規性。

## 主要な貢献 (Contributions)

普遍的関係の発見
- バッチサイズ増加 → 初期は「完璧な比例スケーリング (b-fold benefit)」、次に逓減効果、最終的に「最大有効バッチサイズ (scaling stops)」に到達。
- これは 6種類のモデル、3種類の最適化手法、7つのデータセット すべてに共通。

最大有効バッチサイズの決定要因を分析
- モデルや最適化アルゴリズムに大きく依存（例：Momentum を用いるとより大きなバッチを活用可能）。
- データセットの影響は比較的小さい。

ハイパーパラメータ調整の重要性
- 「学習率はバッチサイズに比例して増やす」という経験則は一般化できない。
- バッチサイズごとに独立に学習率・モーメンタムを最適化する必要。

文献の対立を整理
- 「大きなバッチは汎化を悪化させる」との主張は、メタパラ調整や計算予算の制約によるものと説明。
- 適切な調整をすれば、大きなバッチでも汎化性能は劣化しない。

## 実験方法
- 対象データセット: MNIST, CIFAR-10, ImageNet, Open Images, LM1B, Common Crawl など（表1, p.14）。
- 対象モデル: ResNet-8/50, VGG-11, Transformer, LSTM など（表2, p.14）。
- 最適化手法: SGD, Momentum SGD, Nesterov momentum。
- 評価指標: 各 workload ごとに「目標誤差 (validation error/cross entropy) に到達するまでの学習ステップ数」。
- ハイパーパラ調整: バッチサイズごとに独立に quasi-random search。

### Figure 1 (p.16): すべてのタスクで共通して「初期の比例スケーリング → 逓減効果 → 利得ゼロ」という曲線を描くことを確認。

<img width="614" height="714" alt="Image" src="https://github.com/user-attachments/assets/7c4bf859-cede-4bfe-8cf2-4e605455046e" />
<img width="976" height="389" alt="Image" src="https://github.com/user-attachments/assets/2f0c87ff-670d-42b2-8076-694f131951ea" />

### Figure 3 (p.18): モデル構造の違いにより利用可能なバッチサイズ上限が異なる（ResNet-50 > VGG-11、Transformer > LSTM）。

<img width="563" height="773" alt="Image" src="https://github.com/user-attachments/assets/ae2be6c1-ca26-4046-8a14-4c61172ad0e3" />

### Figure 4 (p.20): Momentum により「完璧なスケーリング領域」が大きなバッチまで拡張される。

<img width="956" height="437" alt="Image" src="https://github.com/user-attachments/assets/176ffe0d-922b-415a-b5a0-b8a2567af073" />

### Figure 11 (p.27): 性能はバッチサイズそのものよりも 計算予算 (steps vs epochs) に強く依存する。
step 予算 → 大きなバッチが有利
epoch 予算 → 小さなバッチが有利

<img width="675" height="830" alt="Image" src="https://github.com/user-attachments/assets/572d8d16-122a-4072-b275-27ab89724f93" />
